rm/rm-server/src/main/java/org/ow2/proactive/resourcemanager/nodesource/infrastructure/HostTracker.java

```diff
@@ -0,0 +1,232 @@
+ * Contact: contact@activeeon.com
+ *
+ * This library is free software: you can redistribute it and/or
+ * modify it under the terms of the GNU Affero General Public License
+ * as published by the Free Software Foundation: version 3 of
+ * the License.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ *
+ * If needed, contact us to obtain a release under GPL Version 2 or 3
+ * or a different license than the AGPL.
+ */
+package org.ow2.proactive.resourcemanager.nodesource.infrastructure;
+
+import java.io.Serializable;
+import java.net.InetAddress;
+import java.util.*;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.concurrent.atomic.AtomicInteger;
+
+import org.apache.log4j.Logger;
+
+
+/**
+ * This class tracks the information related to node deployment on a host.
+ * It is used by {@link HostsFileBasedInfrastructureManager} and objects of
+ * this class are persisted along with the node source data.
+ *
+ * All non-final fields of this class should be thread-safe.
+ */
+public class HostTracker implements Serializable {
+
+    private static final long serialVersionUID = 1L;
+
+    protected static final Logger logger = Logger.getLogger(HostTracker.class);
+
+    private final String hostInFile;
+
+    private final int configuredNodeNumber;
+
+    private final InetAddress host;
+
+    private Map<String, NodeStatus> nodeStatusPerNodeUrl;
+
+    private AtomicInteger aliveNodeCounter;
+
+    private AtomicInteger downNodeCounter;
+
+    private AtomicInteger removedNodeCounter;
+
+    private AtomicBoolean needNodesFlag;
+
+    protected HostTracker(String hostInFile, int configuredNodeNumber, InetAddress host) {
+        this.hostInFile = hostInFile;
+        this.configuredNodeNumber = configuredNodeNumber;
+        this.host = host;
+        nodeStatusPerNodeUrl = new HashMap<>();
+        aliveNodeCounter = new AtomicInteger(0);
+        downNodeCounter = new AtomicInteger(0);
+        removedNodeCounter = new AtomicInteger(0);
+        // a newly created host tracker requires nodes because it has not
+        // been deployed yet
+        needNodesFlag = new AtomicBoolean(true);
+    }
+
+    /**
+     * @return the host as it is written in the hosts file supplied by the user
+     */
+    protected String getHostInFile() {
+        return hostInFile;
+    }
+
+    /**
+     * @return the {@link InetAddress} representation of the host
+     */
+    protected InetAddress getHost() {
+        return host;
+    }
+
+    /**
+     * @return whether this host is marked as requiring nodes
+     */
+    protected boolean getNeedNodesFlag() {
```

Comment:
The semantic and name of these two methods is not clear

Reply:
I removed `Flag` in the names of methods as to have `needsNodes` and `setNeedsNodes`